nixos/modules/services/networking/hostapd.nix

```diff
@@ -1330,6 +1330,21 @@ in
         assertion = cfg.radios != { };
         message = "At least one radio must be configured with hostapd!";
       }
+      {
+        assertion =
+          let
+            wirelessEnabled = config.networking.wireless.enable;
+            wirelessInterfaces = config.networking.wireless.interfaces;
+            hostapdInterfaces = lib.attrNames cfg.radios;
+            hasInterfaceConflict = lib.intersectLists hostapdInterfaces wirelessInterfaces != [ ];
+          in
+          !wirelessEnabled || (wirelessInterfaces != [ ] && !hasInterfaceConflict);
```

Comment:
Can we add a note to `wirelessInterfaces != [ ]`? I think it's not trivial to understand just from reading this that wpa_supplicant will match all interfaces if the list is empty.

Reply:
sure, i just found out by accident - so it might be a good idea to explizite document it again